[FE] 상세 리뷰 페이지 : url router 연동, 변경된 디자인 반영 및 서버에서 api를 연동

BadaHertz52 opened the pull request:
- resolves #71

---

### 🚀 어떤 기능을 구현했나요 ?
- 변경된 피그마 디자인 반영
- 목록이든 url을 통한 이동이든 url을 분석해 router로 상세 리뷰 페이지로 이동
- url을 분석해 서버에 리뷰 데이터 요청

### 🔥 어떻게 해결했나요 ?

### 📝 어떤 부분에 집중해서 리뷰해야 할까요?
- 현재 목서버가 준비되어 있지않아서 api 연동은 코드로만 판단할 수 있어요. 
- 목 데이터가 들어있어서 목데이터를 통해서 확인하고 싶다면 fetch부분을 주석 처리하면 돼요.
-  피그마 시안과 비교해 UI가 잘 만들어졌는지 확인해주세요
-  리뷰  작성 페이지와 공통으로 쓰일 수 있는 컴포넌트들은 common폴더 하위에 만들었어요. 해당 컴포넌트들의 재사용성 부분을 확인해주세요.


### 📚 참고 자료, 할 말

BadaHertz52 committed:
fix: stylelint를 통한 css 속성 정렬 기능 오류 수정

- stylelint 버전16과 충돌되는 플러그인 삭제 : stylelint-config-prettier, stylelint-prettier
- css 정렬에 필요하지 않은 플러그인 삭제 : stylelint-config-standard, stylelint-config-styled-componented, stylelint-webpack-plugin
- 추가로 설치한 플러그인: postcss-syntax, @stylelint/postcss-css-in-js
- stylelint 적용 script 추가
- .stylelintrc.json 수정 : css 관련 rule 설정

BadaHertz52 committed:
refactor: stylelint 적용에 따른 css 속성 정렬

BadaHertz52 committed:
Merge branch 'feat/71-detailed_review_page' of https://github.com/woowacourse-teams/2024-review-me into fe/ci/58-stylelint

BadaHertz52 committed:
fix : 절대 경로 사용 시 오류 수정

오류 : eslintimport/no-unresolved

BadaHertz52 committed:
Merge branch 'develop' of https://github.com/woowacourse-teams/2024-review-me into fe/fix/79-eslint_import_bug_fix

BadaHertz52 committed:
chore: eslintrc.cjs 에서 불필요한 코드 삭제

 node 환경 setting 삭제

BadaHertz52 committed:
style: eslint 적용에 따른 리뷰 상세페이지 import 순서 정리

BadaHertz52 committed:
refactor: formatDate를 utils/date 파일로 이동

BadaHertz52 committed:
design: theme에 colors,, fontSize 변경 및 borderRadius 추가

BadaHertz52 committed:
feat: MultilineTextViewer 컴포넌트 생성

- 개행이 포함된 string에 개행을 적용해서 보여주는 컴포넌트

BadaHertz52 committed:
feat: 깃허브 저장소 이미지 컴포넌트 생성

BadaHertz52 committed:
feat: 리뷰와 관련된 날짜 UI 컴포넌트 생성

BadaHertz52 committed:
featr: LockButton 삭제 LockToggle 추가

BadaHertz52 committed:
refactor: 피그마 디자인 변경에 따른 ReviewDescription 변경

BadaHertz52 committed:
feat: ReviewComment 컴포넌트 생성

BadaHertz52 committed:
refactor: ReviewViewSection -> ReviewSection 으로 변경 및 리팩토링

- 불필요한  컴포넌트 삭제 : RevieAnswer , ReviewQuestion

BadaHertz52 committed:
refactor: DetailedReviewPage 리팩토링

- 목데이터 변경
-  추가 및 변경된 컴포넌트를 사용해 리뷰 상세페이지 컴포넌트(DetailedReviewPage) 리팩토링
- DetailedReviewPage 폴더의 styles.ts 삭제

BadaHertz52 committed:
refactor: review에 대한 타입 변경

BadaHertz52 committed:
design : ReviewDate의 클론 스타일 적용

BadaHertz52 committed:
feat: KeywordSection 컴포넌트 생성

- 리뷰 상세 페이지 키워드 부분 컴포넌트 생성

BadaHertz52 committed:
feat: ReviewSectionHeader 컴포넌트 생성 및 적용

- 리뷰 상세보기에서 반복되는 질문,키워드 헤더부분을 컴포넌트로 분리

BadaHertz52 committed:
design : 리뷰 상세페이지에 width 변경

BadaHertz52 committed:
refactor: DetailedReview의 목데이터 변경 및 리팩토링

- 타입 변경에 따른 목 데이터 변경
- KeywordSection 적용

BadaHertz52 committed:
design : formWidth를 theme에 추가 및 리뷰 작성/리뷰 상세 페이지에 적용

BadaHertz52 committed:
fix: Layout에서 가로 스크롤 생기는 오류 수정

- 100vw는 스크롤을 포함한 뷰포트 너비라서 100%으로 수정

BadaHertz52 committed:
feat: 리뷰 상페이지 router에 라우터 파라미터 적용 및 관련 설정 변경

- 데모데이를 위해 현재 데이터베이스에 있는 리뷰 상세페이지 id를 sidebar의 리뷰 상세페이지 메뉴 link에 적용
- 리뷰 상세페이지(DetailedReviewPage)의 api 핸들러 수정

BadaHertz52 committed:
docs: 변수명 변경 (isLock -> isPublic)

BadaHertz52 committed:
refactor: 깃헙 저장소 로고 주소 변수명 변경

- projectImgSrc -> thumbnailUrl

chysis commented on frontend/src/components/common/ReviewDate/styles.ts:
```diff
@@ -10,8 +10,8 @@ export const ReviewDate = styled.div`
-  span:nth-child(2) {
-    margin: 0 1rem;
-  }
+export const Colon = styled.span`
+  margin: 0 1rem;
```
nth-child 말고 별도의 named component로 지정한 부분 좋습니다👍

chysis reviewed (approved):
바다! 작업하느라 고생 많으셨어요 😊
리팩토링까지 잘 해주셔서 별도로 남길 코멘트는 없는 것 같아요.
Approve 했습니다~

ImxYJL commented on frontend/src/components/common/MultilineTextViewer/index.tsx:
```diff
@@ -0,0 +1,20 @@
+  text: string;
+}
+
+const MultilineTextViewer = ({ text }: MultilineTextViewerProps) => {
```
긴 글을 줄바꿈해주는 기능을 컴포넌트로 만들어야 하는 이유가 궁금합니다! 훅으로 만들었을 때보다 보다 직관적으로 사용(훅의 반환값을 가져와서 별도의 가공을 수행하는 대신 텍스트만 넘겨주면 자동으로 줄바꿈이 되어 렌더링됨)할 수 있어서인가요?

BadaHertz52 replied:
리뷰 목록의 리뷰 카드, 고정된 리뷰에서도 답변이 들어가요. 줄바꿈 기능이 있는 컴포넌트가 있다면  답변이 들어가는 곳에서 재사용할 수 있다고 생각했어요

바쁘다보니 훅으로 빼는거는 생각해보지 못했네요!

ImxYJL commented on frontend/src/components/common/ProjectImg/index.tsx:
```diff
@@ -0,0 +1,16 @@
+const ProjectImg = ({ thumbnailUrl, projectName, $size }: ProjectImgProps) => {
+  const src = thumbnailUrl ?? GithubLogoIcon;
+  const alt = thumbnailUrl ? `${projectName} 저장소 이미지` : '깃허브 로고';
+  return <S.Img src={src} alt={alt} $size={$size} />;
```
사이즈는 재사용성을 위해 별도로 열어둔 건지 궁금해용
그리고 `size` 대신 보다 자세한 변수명은 어떤가요? 현재 width와 height 크기 지정에 사용되고 있던데 이런 의미를 보다 직관적으로 드러내면 어디에 쓰이는지 한 번에 이해하기 쉬울 것 같습니다.
혹시 추후 스타일 확장성을 고려해 추상적으로 지은 걸까요?

BadaHertz52 replied:
로고 이미지는 원이라 너비와 높이가 같아요
너비나 높이로만 props 받자니  둘 모두에 사용하고,
둘 다 props를 받는거는 불필요했어요

지름(diameter)를 쓸까했지만 자주 사용되는 props명 같지는 않았어요 

size라면 너비,높이를 다 포함할 수 있으면서 한 눈에 무슨 의미인지 쉽게 파악할 수 있지않을까 생각했어요

디자인이 다 확정된다면 필요한 로고 사이즈를 상수처리해서 사용할 수 도 있어요

ImxYJL commented on frontend/src/components/common/index.tsx:
```diff
@@ -1 +1,5 @@
 export { default as SearchInput } from './SearchInput';
+export { default as ProjectImg } from './ProjectImg';
```
common 컴포넌트들을 위한 index 파일 좋아요!!

ImxYJL commented on frontend/src/components/layouts/PageLayout/styles.ts:
```diff
@@ -1,7 +1,7 @@
 import styled from '@emotion/styled';
 
 export const Layout = styled.div`
-  width: 100vw;
```
vw -> %로 변경된 이유가 궁금해용

BadaHertz52 replied:
가로 스크롤바가 생기는 이슈가 있었어요
찾아보니 100vw는 스크롤바까지 포함한 뷰포트 너비라   
100vw로 인해 너비가 스크롤바  영역까지 넓혀져 생긴 오류였어요.  %는 부모를 기준으로 하는데 부모에 관련 css 설정이 없다면 타고 올라가 뷰포트가 기준이 되는데 이때 뷰포트 영역은 스크롤바를 제외한 그 안쪽이라 가로 스크롤바 이슈가 발생하지 않아요

단, PageLayout 위의 dom의 width를 설정하지 말아야한다는 제약이 생기니 조심해야 해요

ImxYJL commented on frontend/src/pages/DetailedReviewPage/components/KeywordSection/index.tsx:
```diff
@@ -0,0 +1,26 @@
+}
+const KEY_WORD_HEADER = '키워드';
+
+const KeywordSection = ({ keywords, index }: KeywordSectionProps) => {
```
이 컴포넌트의 이름이 -Section인 이유는 스타일이 적용된 컴포넌트(Container, Wrapper)와의 차별점을 두기 위함이겠죵?

BadaHertz52 replied:
키워드도 하나의 질문과 답변이라고 생각했어요

키워드 위에서 질문과 답변을 하나의 section으로 묶어두어서 키워드도 section을 사용하는 게 시멘틱을 일관되게 사용하는거라 생각했어요

ImxYJL commented on frontend/src/pages/DetailedReviewPage/components/LockToggle/index.tsx:
```diff
@@ -0,0 +1,47 @@
+
+interface LockToggleProps extends Omit<ToggleButtonProps, 'name'> {}
+
+const LockToggle = (props: LockToggleProps) => {
```
Toggle이 명사이기도 하군요...덕분에 알아갑니다ㅋㅋㅋㅋ

BadaHertz52 replied:
명사까지는 깊게 생각안했는데 저도 배워갑니다 😁

ImxYJL commented on frontend/src/pages/DetailedReviewPage/components/LockToggle/index.tsx:
```diff
@@ -0,0 +1,47 @@
+  },
+};
+interface ToggleButtonProps {
+  name: keyof typeof IMAGE;
```
토글 타입을 객체로 만들고 명시적으로 써 준 게 인상깊네요
바다한테 타스 많이 배워가야지~~

ImxYJL reviewed (approved):
코멘트는 쫌 있지만... 거의 개인적인 질문들이라 어프루브합니다ㅎㅎ 일정 촉박했는데 이것저것 리팩토링하고 구현하느라 고생 많았어요!!

BadaHertz52 committed:
Merge branch 'develop' of https://github.com/woowacourse-teams/2024-review-me into fe/feat/71-detailed_review_page

BadaHertz52 committed:
refactor: 리뷰 상세 페이지 api 변경에 따른 수정

- 리뷰 상세 페이지 keyword 타입 변경
- api endpoint 변경 , router 변경, sidebar 상세보기 경로 변경
- useState에 빈 값에 대한 타입 추론 사용

soosoo22 reviewed (approved):
추후에 코드 리뷰 진행할게요!
